demos/imagenet_classification_demo/main.cpp

```diff
@@ -0,0 +1,390 @@
+                        << "Failed to query the metric for the " << FLAGS_d << " with error:" << ex.what();
+            }
+        }
+
+        // ---------------------------------------Create infer request----------------------------------------
+        std::vector<InferenceEngine::InferRequest> inferRequests;
+        int infReqNum = FLAGS_nireq;
+        for(int infReqID = 0; infReqID < infReqNum; ++infReqID)
+            inferRequests.push_back(executableNetwork.CreateInferRequest());
+
+        std::list<cv::Mat> showMats;
+        
+        std::condition_variable condVar;
+        std::mutex mutex;
+        std::mutex showMutex;
+
+        // curImg is stored in ieWrapper
+        unsigned batchSize = FLAGS_b;
+        int irFirstIndex = 0;
+        int modelInputResolution = input_shapes[input_name][2];
+        
+        // --------------------------------------Set completion callback---------------------------------------
+        for (InferenceEngine::InferRequest& inferRequest : inferRequests) {
+            inferRequest.SetCompletionCallback(
+                InferRequestCallback(
+                    inferRequest,
+                    modelInputResolution,
+                    irFirstIndex,
+                    mutex,
+                    condVar,
+                    inputImgs,
+                    inputBlobName,
+                    FLAGS_b,
+                    showMats,
+                    curPos,
+                    framesNum,
+                    quitFlag
+                )
+            );
+            irFirstIndex = (irFirstIndex + batchSize) % inputImgs.size();
+        }
+        
+        // -------------------------------------------Filling blobs-------------------------------------------
+        auto blobDims = inputBlobsDimsInfo[inputBlobName];
+        if (blobDims.size() != 4) {
+            throw std::runtime_error("Input data does not match size of the blob");
+        }
+
+        int64 startTime = cv::getTickCount();
+        for (InferenceEngine::InferRequest& inferRequest : inferRequests) {     
+            auto inputBlob = inferRequest.GetBlob(inputBlobName);
+
+            for (unsigned i = 0; i < batchSize; i++) {
+                int imageIndex = (curPos + i) % inputImgs.size();
+                cv::Mat inputImg = inputImgs[imageIndex];
+                resizeImage(inputImg, modelInputResolution);
+
+                matU8ToBlob<uint8_t>(inputImg, inputBlob, i);
+            }
+            curPos = (curPos + batchSize) % inputImgs.size();
+
+            inferRequest.StartAsync();
+        }
+
+        // ----------------------------------------Create output info-----------------------------------------
+        int width;
+        int height;
+        std::vector<std::string> gmRowsCols = split(FLAGS_res, 'x');        
+        if (gmRowsCols.size() != 2) throw std::runtime_error("Input gmsizes key is not valid.");
```

Comment:
The message should be clarified, it will not be clear what `gmsizes` is for people who don't read the code.

Reply:
Done